Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml

```diff
@@ -15,27 +15,95 @@ tactics: []
+  | extend ThreatPurpose=findingTokens[0], findingTokens=split(findingTokens[1], "/")
+  | extend ResourceTypeAffected=findingTokens[0], findingTokens= split(findingTokens[1], ".")
+  | extend ThreatFamilyName=findingTokens[0], findingTokens=split(findingTokens[1], "!")
+  | extend DetectionMechanism=findingTokens[0], Artifact=findingTokens[1]
   | extend UniqueFindingId = Id
   | extend AWSAcoundId = AccountId
-  | project-away tokens,ActivityType, Id, AccountId
-  | project-away TimeGenerated, TenantId, SchemaVersion, Region, Partition
-  | extend Severity= iff(Severity between (7.0..8.9),"High",iff(Severity between
-  (4.0..6.9), "Medium", iff(Severity between (1.0..3.9),"Low","Unknown")))
-version: 1.0.2
+  | extend Severity=case(
+    Severity > 7.0, "High",
+    Severity between (4.0 .. 6.9), "Medium",
+    Severity between (1.0 .. 3.9), "Low",
+    "Unknown"
+    )
+  // Pull out resource details we can extract entities from
+  // https://docs.aws.amazon.com/guardduty/latest/APIReference/API_Resource.html
+  | extend ResourceDetailsObject=case(
+    isnotempty(ResourceDetails.accessKeyDetails), ResourceDetails.accessKeyDetails,
+    isnotempty(ResourceDetails.containerDetails), ResourceDetails.containerDetails,
+    isnotempty(ResourceDetails.ebsVolumeDetails), ResourceDetails.ebsVolumeDetails,
+    isnotempty(ResourceDetails.ecsClusterDetails), ResourceDetails.ecsClusterDetails,
+    isnotempty(ResourceDetails.eksClusterDetails), ResourceDetails.eksClusterDetails,
+    isnotempty(ResourceDetails.instanceDetails), ResourceDetails.instanceDetails,
+    isnotempty(ResourceDetails.kubernetesDetails), ResourceDetails.kubernetesDetails,
+    isnotempty(ResourceDetails.lambdaDetails), ResourceDetails.lambdaDetails,
+    isnotempty(ResourceDetails.rdsDbInstanceDetails), ResourceDetails.rdsDbInstanceDetails,
+    isnotempty(ResourceDetails.rdsDbUserDetails), ResourceDetails.rdsDbUserDetails,
+    isnotempty(ResourceDetails.s3BucketDetails), ResourceDetails.s3BucketDetails,
+    dynamic(null)
+    )
+  // Pull out service action details we can extract entities from
+  // https://docs.aws.amazon.com/guardduty/latest/APIReference/API_Action.html
+  | extend ActionDetailsObject=case(
+    isnotempty(ServiceDetails.action.awsApiCallAction), ServiceDetails.action.awsApiCallAction,
+    isnotempty(ServiceDetails.action.dnsRequestAction), ServiceDetails.action.dnsRequestAction,
+    isnotempty(ServiceDetails.action.kubernetesApiCallAction), ServiceDetails.action.kubernetesApiCallAction,
+    isnotempty(ServiceDetails.action.networkConnectionAction), ServiceDetails.action.networkConnectionAction,
+    isnotempty(ServiceDetails.action.portProbeAction), ServiceDetails.action.portProbeAction,
+    isnotempty(ServiceDetails.action.rdsLoginAttemptAction), ServiceDetails.action.rdsLoginAttemptAction,
+    dynamic(null)
+    )
+  | extend RemoteIpAddress = max_of(
+    tostring(ServiceAction.remoteIpDetails.ipAddressV4),
```

Comment:
There is no entity with the same ServiceAction as everything was renamed as ActionDetailsObject in line 49 to 56.